ComputeClient/Compute.Client.UnitTests/Server20/ServerAccessorTests.cs

```diff
@@ -47,6 +49,33 @@ public async Task ListServers_ReturnsResponse()
             Assert.AreEqual("QA1_N1_VMWARE_1", servers.First().datacenterId);
         }
 
+        [TestMethod]
+        public async Task ListServersWithFilterOptions_ReturnsResponse()
+        {
+            var serverListOptions = new ServerListOptions
+            {
+                Filters =
+                {
+                    new Filter { Field = ServerListOptions.StateField,Value = "PENDING_ADD" },
+                    new Filter { Field = ServerListOptions.SourceTypeField,Value = "SERVER_SNAPSHOT_ID" },
+                    new Filter { Field = ServerListOptions.SourceValueField,Value="c4e80cb1-e819-4dbf-97aa-020a7f1d9fd3" }
+                }
+            };
+
+            var queryString = "state=PENDING_ADD&sourceType=SERVER_SNAPSHOT_ID&sourceValue=c4e80cb1-e819-4dbf-97aa-020a7f1d9fd3";
+            var listServersUri = new Uri(ApiUris.ListServers(accountId) + "?" + queryString, UriKind.Relative);
+
+            requestsAndResponses.Add(listServersUri, RequestFileResponseType.AsGoodResponse("ListServersResponse.xml"));
+
+            var client = GetWebApiClient();
+            var accessor = new ServerAccessor(client);
+            var servers = await accessor.ListServers(serverListOptions);
```

Comment:
the _requestsAndResponses_ is a dictionary where the URI is used as the key.

So its being used here to verify that URI built using the _serverListOptions_ matches the expected _listServersUri_ above (this will ultimately be the URL that is sent to the  Cloud Control API).

Ultimately we should see some url in the format **API_HTTPS_URL/caas/2.11/a3d9aacc-a273-45a5-919d-0f4c41c0763b/server/server?sourceType=someSourceType&sourceValue=someSourceValue**